src/parser/creative_companion_parser.js

```diff
@@ -14,79 +14,89 @@ import { parserUtils } from './parser_utils';
 
-      parserUtils
-        .childrenByName(companionResource, 'HTMLResource')
-        .forEach(htmlElement => {
-          companionAd.type =
-            htmlElement.getAttribute('creativeType') || 'text/html';
-          companionAd.htmlResource = parserUtils.parseNodeText(htmlElement);
-        });
+      companionAd.htmlResource =
+        parserUtils.parseNodeText(
+          parserUtils.childByName(companionResource, 'HTMLResource')
+        ) || null;
 
-      parserUtils
-        .childrenByName(companionResource, 'IFrameResource')
-        .forEach(iframeElement => {
-          companionAd.type = iframeElement.getAttribute('creativeType') || 0;
-          companionAd.iframeResource = parserUtils.parseNodeText(iframeElement);
-        });
+      companionAd.iframeResource =
+        parserUtils.parseNodeText(
+          parserUtils.childByName(companionResource, 'IFrameResource')
+        ) || null;
 
-      parserUtils
-        .childrenByName(companionResource, 'StaticResource')
-        .forEach(staticElement => {
-          companionAd.type = staticElement.getAttribute('creativeType') || 0;
-
-          parserUtils
-            .childrenByName(companionResource, 'AltText')
-            .forEach(child => {
-              companionAd.altText = parserUtils.parseNodeText(child);
-            });
+      const staticElement = parserUtils.childByName(
+        companionResource,
+        'StaticResource'
+      );
+      if (staticElement) {
+        companionAd.staticResource = parserUtils.parseNodeText(staticElement);
+        companionAd.creativeType =
+          staticElement.getAttribute('creativeType') || null;
 
-          companionAd.staticResource = parserUtils.parseNodeText(staticElement);
-        });
+        const altTextElement = parserUtils.childByName(
+          companionResource,
+          'AltText'
+        );
+        companionAd.altText = parserUtils.parseNodeText(altTextElement) || null;
+      }
 
-      parserUtils
-        .childrenByName(companionResource, 'TrackingEvents')
-        .forEach(trackingEventsElement => {
-          parserUtils
-            .childrenByName(trackingEventsElement, 'Tracking')
-            .forEach(trackingElement => {
-              const eventName = trackingElement.getAttribute('event');
-              const trackingURLTemplate = parserUtils.parseNodeText(
-                trackingElement
-              );
-              if (eventName && trackingURLTemplate) {
-                if (!Array.isArray(companionAd.trackingEvents[eventName])) {
-                  companionAd.trackingEvents[eventName] = [];
-                }
-                companionAd.trackingEvents[eventName].push(trackingURLTemplate);
+      const trackingEventsElement = parserUtils.childByName(
+        companionResource,
+        'TrackingEvents'
+      );
+      if (trackingEventsElement) {
+        parserUtils
+          .childrenByName(trackingEventsElement, 'Tracking')
+          .forEach(trackingElement => {
```

Comment:
why are we using `forEach` here and `map` on line 21?

Reply:
because `companionAd.trackingEvents` is an object not an array

Reply:
Not sure I get what you mean, both times you chain to `childrenByName` which always returns an array.